contracts/src/protocol/OmniAVS.sol

```diff
@@ -0,0 +1,227 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.12;
+
+import { DelegationManager } from "eigenlayer-contracts/src/contracts/core/DelegationManager.sol";
+import { ISignatureUtils } from "eigenlayer-contracts/src/contracts/interfaces/ISignatureUtils.sol";
+import { IStrategy } from "eigenlayer-contracts/src/contracts/interfaces/IStrategy.sol";
+import { IDelegationManager } from "eigenlayer-contracts/src/contracts/interfaces/IDelegationManager.sol";
+
+import { OperatorStateRetriever } from "eigenlayer-middleware/src/OperatorStateRetriever.sol";
+import { ServiceManagerBase } from "eigenlayer-middleware/src/ServiceManagerBase.sol";
+import { RegistryCoordinator } from "eigenlayer-middleware/src/RegistryCoordinator.sol";
+import { IRegistryCoordinator } from "eigenlayer-middleware/src/interfaces/IRegistryCoordinator.sol";
+import { IStakeRegistry } from "eigenlayer-middleware/src/interfaces/IStakeRegistry.sol";
+
+import { OmniPredeploys } from "../libraries/OmniPredeploys.sol";
+import { IOmniPortal } from "../interfaces/IOmniPortal.sol";
+import { IOmniEthRestaking } from "../interfaces/IOmniEthRestaking.sol";
+import { IOmniAVS } from "../interfaces/IOmniAVS.sol";
+import { IOmniAVSAdmin } from "../interfaces/IOmniAVSAdmin.sol";
+
+/**
+ * @title OmniAVS
+ * @notice Omni AVS contract. It is responsible for syncing Omni AVS operator
+ *         stake and delegations with the Omni chain.
+ */
+contract OmniAVS is IOmniAVS, IOmniAVSAdmin, ServiceManagerBase, OperatorStateRetriever {
+    /// @dev AVS Quorum numbers, Omni only has one quorum
+    bytes public constant QUORUM_NUMBERS = hex"00";
+
+    /// @dev AVS Quorum number
+    uint8 public constant QUORUM_NUMBER = 0;
+
+    /// @dev Omni chain id, used to make xcalls to the Omni chain
+    uint64 public omniChainId;
+
+    /// @dev Omni portal contract, used to make xcalls to the Omni chain
+    IOmniPortal public omni;
+
+    /// @dev List of currently register operators, used to sync EigenCore
+    address[] public operators;
+
+    /// @dev Sync with OmniAVS StakeRegistry with EigenCore. To be called before any
+    //       operation that requires an up-to-date view of operator stake.
+    modifier syncWithEigenCore() {
+        RegistryCoordinator(address(_registryCoordinator)).updateOperators(operators);
```

Comment:
can we make this not a modifier? Looking at `xFeeForSync()` and `syncWithOmni()` (especially the first), it's not obvious that it changes state. In general, I think it's an anti-pattern to have a modifier that changes state anyway, it should really just be for authorization. Modifiers hide code

Reply:
yeah i like that